core/state/statedb.go

```diff
@@ -530,6 +530,14 @@ func (s *StateDB) Selfdestruct6780(addr common.Address) {
 	}
 }
 
+func (s *StateDB) WasCreatedInCurrentTx(addr common.Address) bool {
+	stateObject := s.getStateObject(addr)
+	if stateObject == nil {
+		return false
+	}
+	return stateObject.created
+}
```

Comment:
Quite unfortunate that we need to add an extra method to StateDB, but it's a simple one. 
Open to renaming the method if sounds too verbose.